repository/sql.go

```diff
@@ -99,6 +99,58 @@ func (sql *sqlRepository) CreateAsset(assetData oam.Asset) (*types.Asset, error)
 	}, nil
 }
 
+// DeleteAsset removes an asset in the database by its ID.
+// It takes a string representing the asset ID and removes the corresponding asset from the database.
+// Returns an error if the asset is not found.
+func (sql *sqlRepository) DeleteAsset(id string) error {
+	if rels, err := sql.IncomingRelations(&types.Asset{ID: id}); err == nil {
+		for _, rel := range rels {
+			if err := sql.DeleteRelation(rel.ID); err != nil {
+				return err
+			}
+		}
+	}
+
+	if rels, err := sql.OutgoingRelations(&types.Asset{ID: id}); err == nil {
+		for _, rel := range rels {
+			if err := sql.DeleteRelation(rel.ID); err != nil {
+				return err
+			}
+		}
+	}
```

Comment:
in the future, may want to add a non-public method to get all relations in a single query. fine the way it is for now.

Reply:
I definitely agree. The idea crossed my mind while implementing the code above